samples/interactive-tutorials/search/search-with-boost-spec.js

```diff
@@ -14,24 +14,54 @@
 
 'use strict';
 
-async function main() {
+async function main(impersonatedPrincipal) {
   // [START retail_search_product_with_boost_spec]
   // Call Retail API to search for a products in a catalog, rerank the
   // results boosting or burying the products that match defined condition.
 
+  const { GoogleAuth, Impersonated } = require('google-auth-library');
+
+  // Acquire source credentials:
+  const auth = new GoogleAuth();
+  const client = await auth.getClient();
+  const projectId = await auth.getProjectId();
+
+  let opts = {};
+
+  if(impersonatedPrincipal) {
```

Comment:
I'm not sure if this is gonna work. Our API requires a SA associated with a project for quota checking / billing etc., which won't work with personal credentials. There should be ways to file exemptions to allow your test project to export SA keys.

Reply:
Yes, it works, this feature is added for some organisations who may disable  "service account key export" in their GCP IAM policy as this is addressed not recommended by Google. If they do so, then the setup script in the interactive tutorial for environment preparation to run all the api calls to the Retail API won't work, so that the tutorials may provide a way to avoid exporting key file from service account, which is suggested by Google that using "Impersonated service account" can resolve that.

https://cloud.google.com/iam/docs/impersonating-service-accounts

In fact, not only the API calls from the source code needs to be changed, but also the interactive tutorial it self. 
Next setup, maybe I will add a util function in setup folder,  to centralize the auth process.